### PR TITLE
Add backupinfra listing support

### PR DESCRIPTION
**Pull request by swapnilgm**
Signed-off-by: Swapnil Mhamane <swapnil.mhamane@sap.com>

**What this PR does / why we need it**:
This PR add the support to list backupInfrastructure resources under projects.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```noteworthy operator
Added support for listing `BackupInfrastructure` resource under project folder
```

### CODE REVIEW

File: src/gardener/gardener-tree.js [hunk 1]
```diff
@@ -195,10 +205,15 @@ function projectTooltip (element) {
 }
 
 function clusterTypes (project) {
```
Comment (petersutter) [inline src/gardener/gardener-tree.js]: rename to something like `projectResources`, `projectScopedResources` or maybe you have a better name.. Initially the list contained only cluster-like resources (plant, shoot) but this changed so it makes sense to also rename the function

File: src/gardener/gardener-tree.js [hunk 2]
```diff
@@ -280,6 +295,37 @@ function plantTooltip (element) {
...
+
+function toBackupInfraTreeNode (project, backupInfra) {
+  const name = _.get(backupInfra, 'metadata.name')
+  return {
+    nodeType: nodeType.NODE_TYPE_BACKUPINFRA,
+    kindPlural: 'backupinfrastructures',
+    name,
+    project,
+    seed: _.get(backupInfra, 'spec.seed'),
+    forceDeletionEnabled: _.get(backupInfra, ['metadata', 'annotations', 'backupinfrastructure.garden.sapcloud.io/force-deletion']) || "false",
```
Comment (petersutter) [inline src/gardener/gardener-tree.js]: the lodash get function supports a default parameter
```suggestion
    forceDeletionEnabled: _.get(backupInfra, ['metadata', 'annotations', 'backupinfrastructure.garden.sapcloud.io/force-deletion'], "false"),
```

File: src/gardener/gardener-tree.js [hunk 3]
```diff
@@ -37,7 +38,8 @@ const nodeType = {
   NODE_TYPE_LANDSCAPE: 'landscape',
   NODE_TYPE_FOLDER: 'folder',
   NODE_TYPE_CLUSTER_TYPE: 'clusterType',
```
Comment (petersutter) [inline src/gardener/gardener-tree.js]: rename to something like `NODE_TYPE_PROJECT_RESOURCE` or `NODE_TYPE_PROJECT_SCOPED_RESOURCE`

**Review by petersutter — CHANGES_REQUESTED** [review]

**Comment by swapnilgm** [comment]
Thanks you for the review. I have addressed your comments. PTAL.

**Review by petersutter — APPROVED** [review]
lgtm, ty

File: src/gardener/gardener-tree.js [hunk 4]
```diff
@@ -194,11 +204,16 @@ function projectTooltip (element) {
...
 }
 
-function clusterTypes (project) {
-  return [
+function projectResources (project) {
+  const resources = [
     toFolderTreeNode(project, 'Shoots', nodeType.NODE_TYPE_SHOOT),
     toFolderTreeNode(project, 'Plants', nodeType.NODE_TYPE_PLANT)
   ]
+  const canIGetBackupInfras = await accessReview.canI('get', 'backupinfrastructures')
```
Comment (petersutter) [inline src/gardener/gardener-tree.js:213]:
```suggestion
  const accessReview = new SelfSubjectAccessReviewClient(project.landscape.kubeconfig)
  const canIGetBackupInfras = await accessReview.canI('get', 'backupinfrastructures')
```

**Review by petersutter — CHANGES_REQUESTED** [review]
I saw one more thing to add as `accessReview` was not defined (see suggestion), then it will be ok. PTAL

**Comment by swapnilgm** [comment]
Done

**Review by petersutter — APPROVED** [review]
lgtm, ty

**Pull request merged**